instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py

```diff
@@ -358,12 +358,17 @@ def collect_custom_request_headers_attributes(environ):
             OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SANITIZE_FIELDS
         )
     )
+    headers = {}
 
-    headers = {
-        key[_CARRIER_KEY_PREFIX_LEN:].replace("_", "-"): val
-        for key, val in environ.items()
-        if key.startswith(_CARRIER_KEY_PREFIX)
-    }
+    for key, val in environ.items():
+        if key.startswith(_CARRIER_KEY_PREFIX):
+            header_key = (
+                key[_CARRIER_KEY_PREFIX_LEN:].replace("_", "-").lower()
+            )
```

Comment:
I don't think any changes are needed to the WSGI instrumentation. The [WSGI/CGI specs](https://datatracker.ietf.org/doc/html/draft-coar-cgi-v11-03#:~:text=Meta%2Dvariables%20with%20names,onto%20a%20single%20line.) says that HTTP headers should already be canonicalized into upper case with underscores and since it's a dictionary, multiple header values should be merged with `,`

Reply:
For request headers I think you're right, but presumably response headers do need to be changed.